database/internal/treap/immutable.go

```diff
@@ -104,19 +105,87 @@ func (t *Immutable) Get(key []byte) []byte {
 	return nil
 }
 
-// Put inserts the passed key/value pair.
-func (t *Immutable) Put(key, value []byte) *Immutable {
+// Put puts the passed in key/value pairs into the treap.  For operations
+// requiring many insertions at once, Put is memory efficient as the
+// intermediary treap nodes created between each put operation is recycled
+// through an internal sync.Pool, reducing overall memory allocation.
+//
+// If the passed in length of values is less than keys, the keys with no
+// matching values will have nil values.  If the passed in values are nil, all
+// keys will be saved will nil values.
+func (t *Immutable) Put(keys, values [][]byte) *Immutable {
```

Comment:
What about also applying this to `Delete`?

Reply:
We can. In terms of performance it'd be negligible so I didn't bother but I will